octodns_constellix/__init__.py

```diff
@@ -10,7 +10,7 @@
 from collections import defaultdict
 
 from pycountry_convert import country_alpha2_to_continent_code
-from requests import Session
+from requests_cache import CachedSession
```

Comment:
@ross @viranch using this helps to remove lots of caching code that seems to be unused anyway due to the single-shot logic of octodns updates.

Reply:
Only downside I could see to caching at the request layer is that the parsing/munging of the response would have to happen multiple times. I'm not familiar enough with the provider to know what/how it's caching currently.  It does at least seem to be caching domain -> id mappings in `_domains`

Reply:
There is not much parsing/munging work to be done in this provider. The _expensive_ part is the duration of the API calls. In the current version, the cache maintains lists of records and pools. However, the list of pools contains incomplete information. To implement the full-fledged monitoring, it is necessary to fetch detail information for each pool. This makes the "manual" / "homegrown" cache logic superfluous to some extent.